src/App.tsx

```diff
@@ -43,16 +45,21 @@ class App extends React.Component<{}, AppState> {
         });
     }
 
-    public switchSmartContract(newActiveContractLabel: string): void {
+    switchSmartContract(newActiveContractLabel: string): void {
+        const smartContracts: Array<ISmartContract> = this.state.childState.smartContracts;
         this.setState({
             childState: {
-                activeSmartContract: this.state.childState.smartContracts.filter((obj: ISmartContract) => obj.label === newActiveContractLabel)[0],
-                smartContracts: this.state.childState.smartContracts
+                activeSmartContract: smartContracts.filter((obj: ISmartContract) => obj.label === newActiveContractLabel)[0],
```

Comment:
`smartContracts.filter((obj: ISmartContract) => obj.label === newActiveContractLabel)[0]` is the same as:
`smartContracts.find((obj: ISmartContract) => obj.label === newActiveContractLabel)`